### PR TITLE
Sdf

### LINKED ISSUE #689: fix new area implementation in robot_skills
Please apply it in SDF branch, So this can be done parallel to SDF implementation.

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Fixes #689
Merge after: https://github.com/tue-robotics/ed/pull/73

### CODE REVIEW

File: robot_skills/src/robot_skills/util/volume.py [hunk 1]
```diff
@@ -79,14 +82,98 @@ def contains(self, point):
...
+        return self._min_corner.x() <= point.x() <= self._max_corner.x() and \
+               self._min_corner.y() <= point.y() <= self._max_corner.y() and \
+               self._min_corner.z() <= point.z() <= self._max_corner.z()
 
     def __repr__(self):
         return "BoxVolume(min_corner={}, max_corner={})".format(self.min_corner, self.max_corner)
 
 
+class CompositeBoxVolume(Volume):
+    """ Represents a composite box shaped volume """
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/volume.py]: Something which is a composite, so created from multiple things. In this case, multiple boxes.

[the remark this answers is not in the document]
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/util/volume.py]: Then I don't understand the constructor...

[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/volume.py]: See updated docstring

[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/volume.py]: This should be fixed in last commit.

File: robot_skills/src/robot_skills/util/volume.py [hunk 2]
```diff
@@ -101,56 +189,73 @@ def __repr__(self):
         return "OffsetVolume(offset={})".format(self._offset)
 
 
-def volumes_from_entity_info_data(data):
+def volume_from_entity_volume_msg(msg):
     """ Creates a dict mapping strings to Volumes from the EntityInfo data dictionary
 
-    :param data: ed_msgs.msg.EntityInfo.data (string)
+    :param msg: ed_msgs.msg.Volume
     :return: dict mapping strings to Volumes
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/volume.py]: Indeed, fixed in last commit.

File: robot_skills/src/robot_skills/util/volume.py [hunk 3]
```diff
@@ -79,14 +82,96 @@ def contains(self, point):
...
+        convex_hull = [kdl.Vector(min_x, min_y, min_z), kdl.Vector(max_x, min_y, min_z),
+                       kdl.Vector(max_x, max_y, min_z), kdl.Vector(min_x, max_y, min_z)]
+        return convex_hull
+
+    def contains(self, point):
+        """ Checks if the point is inside this volume
+        :param point: kdl Vector w.r.t. the same frame as this volume
+        :return: True if inside, False otherwise
+        """
+        for min_corner, max_corner in zip(self._min_corners, self._max_corners):
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/volume.py]: (nearly) All the code in this class use min and max corners. So zipping it in each function, creates a lot of double code.

**Pull request merged**